src/BetterTeamListTrait.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\apigee_edge_ui;
+
+/**
+ * Copyright (C) 2020 PRONOVIX GROUP BVBA.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Template\Attribute;
+use Drupal\apigee_edge\Element\StatusPropertyElement;
+use Drupal\apigee_edge_teams\Entity\TeamInterface;
+
+/**
+ * List builder additions for teams.
+ */
+trait BetterTeamListTrait {
+
+  /**
+   * Alters the app info in the original list render array.
+   *
+   * @param array $build
+   *   The original render array.
+   * @param string $rel
+   *   The app name's link relationship type, defaults to 'canonical'.
+   */
+  private function buildTeamListContent(array &$build, string $rel = 'canonical'): void {
+    // Use custom template instead of table.
+    unset($build['table']['#type']);
+    if (isset($this->entityTypeId)) {
+      $build['table']['#type'] = $this->entityTypeId;
+    }
+    $build['table']['#theme'] = 'apigee_edge_ui_list';
+
+    // Build the team rows from scratch.
+    $build['table']['#items'] = [];
+    foreach ($this->load() as $entity) {
+      /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $entity */
```

Comment:
1) Trait must not have access to methods/properties that are not defined/required by the trait. `$this->load()` is one of these.
2) Only if the class where this trait is used is a subclass of `\Drupal\apigee_edge_teams\Entity\ListBuilder\TeamListBuilder::load()` you can be sure that the returned team listed is filtered by "view" permission, otherwise, you can't, IOW users may see teams that they should not.